src/api/handlers/guild.ts

```diff
@@ -587,6 +587,36 @@ export async function editEmbed(
   );
 }
 
+/** Returns the widget for the guild. */
+export async function getWidget(guildID: string, options?: { force: boolean }) {
+  const guild = await cacheHandlers.get("guilds", guildID);
+  if (!options?.force) {
+    if (!guild) throw new Error(Errors.GUILD_NOT_FOUND);
+    if (!guild?.widgetEnabled) throw new Error(Errors.GUILD_WIDGET_NOT_ENABLED);
+  }
+
+  return RequestManager.get(`${endpoints.GUILD_WIDGET(guildID)}.json`);
```

Comment:
We should provide a return type for this

Reply:
There is no type documented. But I can try and create custom type for this

Reply:
@itohatweb we **should never** add undocumented stuff to the types.

Reply:
Well then we won't provide a return type for this.

Reply:
You can create an issue on [their repository](https://github.com/discord/discord-api-docs/issues) and wait for the types to be added, till then this PR will be marked as `pending`.

Reply:
Issue created. https://github.com/discord/discord-api-docs/issues/2517